subsys/bluetooth/host/l2cap.c

```diff
@@ -2985,3 +2990,25 @@ int bt_l2cap_chan_send(struct bt_l2cap_chan *chan, struct net_buf *buf)
 	return bt_l2cap_chan_send_cb(chan, buf, NULL, NULL);
 }
 #endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
+
+void l2cap_tx_destroy(void *meta_data)
+{
+	__ASSERT_NO_MSG(meta_data);
+
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
+	if (PART_OF_ARRAY(l2cap_tx_meta_data, (struct l2cap_tx_meta_data *)meta_data)) {
+		struct l2cap_tx_meta_data *data = meta_data;
+
+		if (data->user_data) {
+			att_tx_destroy(data->user_data);
+		}
```

Comment:
It doesn't seem particularly clean or correct to assume that the layer above L2CAP is always ATT. Perhaps we need to have a destroy callback that the higher layer can set for its user data?

Reply:
That is probably cleaner, same with ISO/L2CAP above conn.

Reply:
Added a way to set destroy callbacks in conn and L2CAP. Not sure if it is the best way.